Optionally prevent printing of DRAFT documents

patilsangram opened the pull request:
- Added 'Prevent Draft from Printing' check box in Print Setting. 
- If 'Prevent Draft from printing' is checked then prevent Draft document from printing and generating PDF except for "Administrator".
-  Also same for list view, if user try to print cancelled or Draft doc through list view(If Prevent from Draft checked).
- Reference issue - https://github.com/frappe/erpnext/issues/4688

Thanks,
Sangram Patil
New Indictrans Technologies Pvt Ltd

patilsangram committed:
added checkbox in Print Setting and if checked then draft document prevent from printing and generating pdf

patilsangram committed:
changes for enable print to non submittable document

patilsangram committed:
unable print for cancelled or Draft document if user select in listview to print

rmehta commented on frappe/print/doctype/print_settings/print_settings.json:
```diff
@@ -278,10 +289,37 @@
    "bold": 0, 
    "collapsible": 0, 
+   "description": "Prevent Draft Document from Print and generating PDF", 
+   "fieldname": "prevent_draft_from_printing", 
```
field should be `allow_print_for_draft`

rmehta commented:
@patilsangram thanks!

Can you also make an option for cancelled documents in Print Settings

Also check print in list view

patilsangram commented:
@rmehta  

Sir, 

- I already added code for Cancelled Document on same option. 

- User can't print Cancelled or Draft document from list view, If **Allow Print For Draft** unchecked

- If User try to print cancelled document on form, he got message- **Cannot print cancelled documents**

- But when user Open Draft form the, there is no option form Print unless  **Allow Print For Draft**

- Or shall I provide another option for **Allow Print For Cancelled** in Print Setting

![image](https://cloud.githubusercontent.com/assets/14306547/13914941/985db95c-ef77-11e5-8edd-8e9b26651cbb.png)

![image](https://cloud.githubusercontent.com/assets/14306547/13914906/746a3a2a-ef77-11e5-9acd-c1d255339b78.png)

rmehta commented:
@patilsangram thanks, make a separate option for cancelled. Explict is better than implicit

patilsangram commented:
@rmehta 

Ok sir.

patilsangram commented:
@rmehta 

- As per your suggestion I added another option - **Allow Print for Cancelled** 


- In both option I added one condition that Administrator can **Print**  whether **Allow Print for Draft / Cancelled** or not.

- So condition applied only for other User's **except Administrator**

- Or shall I remove that.

patilsangram committed:
added another option in Print Setting- Allow Print for Cancelled
